Sage/carbon
# fix(multi-select): ensure aria-selected has the correct value

Currently on all selects, including MultiSelect, we set aria-selected=true on only the option that is currently focused. In particular, when using a screenreader and navigating through the options, "selected" is announced for each one. This may be correct for single select (because navigating through an item puts its text value into the select textbox), it is definitely not correct for MultiSelect and means the user has no way of knowing which values are currently selected when using a screenreader.

fix #5272

### Proposed behaviour

The value of aria-selected on options in MultiSelect should depend on whether they are currently selected or not (as shown in the pills inside the textbox).

Also the aria-multiselectable=true should be on the <ul> with role of "listbox"



### Current behaviour

aria-selected on the options of MultiSelect is always false unless currently highlighted (eg by keyboard navigation).

The attribute aria-multiselectable=true is found on the select-list-wrapper div, which is not the right place for it and caused an Axe error.



### Checklist




#### QA


### Additional context



### Testing instructions



Test MultiSelect - with both Option and OptionRow children - with screenreaders (VoiceOver on Mac, NVDA on Windows). Ensure that when reading through the Select options, "selected" or "not selected" (or equivalent) is announced correctly according to whether or not the element is actually currently selected (shown in the pills inside the textbox).

Additionally, make sure the Axe error "elements must only use allowed ARIA attributes which refers to aria-multiselectable - which can be seen on the production storybook docs while the select list is open - does not occur after this PR. (There are still Axe areas under this heading but they are due to storybook itself, not Carbon - in particular none of them should refer to aria-multiselectable.)
